src/Messages/OWSMessageSender.m

```diff
@@ -697,26 +677,12 @@ - (void)groupSend:(NSArray<SignalRecipient *> *)recipients
     [completionFuture catchDo:^(id failure) {
         // failure from toc_thenAll yeilds an array of failed Futures, rather than the future's failure.
         if ([failure isKindOfClass:[NSArray class]]) {
-            NSArray *errors = (NSArray *)failure;
-            for (TOCFuture *failedFuture in errors) {
-                if (!failedFuture.hasFailed) {
-                    // If at least one send succeeded, don't show message as failed.
-                    // Else user will tap-to-resend to all recipients, including those that already received the
-                    // message.
-                    return successHandler();
-                }
-            }
-
-            // At this point, all recipients must have failed.
-            // But we have all this verbose type checking because TOCFuture doesn't expose type information.
-            id lastError = errors.lastObject;
-            if ([lastError isKindOfClass:[TOCFuture class]]) {
-                TOCFuture *failedFuture = (TOCFuture *)lastError;
-                if (failedFuture.hasFailed) {
-                    id failureResult = failedFuture.forceGetFailure;
+            NSArray *groupSendFutures = (NSArray *)failure;
+            for (TOCFuture *groupSendFuture in groupSendFutures) {
```

Comment:
Okay, I've thought it over and there is a solid reason to hang the `isRetryable` state on the error - we want to look at it when resolving possibly multiple errors for a group send into a single error.  If any of them are retryable, we want to retry.  I'm going to add another commit that does this.